src/CasLogic.js

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2015 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+'use strict';
+
+var url = require ('url');
+
+function getServiceUrl() {
+    var parsedUrl = url.parse(window.location.href);
+    return parsedUrl.protocol + "//" + parsedUrl.host + parsedUrl.pathname;
+}
```

Comment:
Just chatted about this with Steve, but we should be aware that the service URL here is the URL of the client rather than the Home Server. Given that CAS providers often whitelist service URLs, they would effectively be whitelisting clients rather than Home Servers, when I think what they'd actually want to do is allow you to log in to a particular home server with any client. This is much more complicated though since we'd have to do another redirect jump via the Home Server, etc.

This also has implications for how we do CAS login on mobile devices (if the HS provided the service URL, this would solve the problem of what service URL we use for a mobile client that otherwise has no web server of its own.)

This is good enough to get us up & running for now though.